core/src/banking_stage.rs

```diff
@@ -899,12 +900,23 @@ impl BankingStage {
     ) -> (usize, Vec<usize>) {
         let mut chunk_start = 0;
         let mut unprocessed_txs = vec![];
+            
+
+        /* TODO - 
+        // instead of using `MAX_NUM_TRANSACTIONS_PER_BATCH` to get chunks, invoking cost_model here to
+        // breaks `transactions` into chunks of re-organized transaction, each complies with cost
+        // model
+        let cost_model = CostModel::new();
+        let chunks = cost_model.pack_transactions_by_cost( transactions );
+        for &chunk in chunks  {
+            reuse below code
+        }
+        // */
```

Comment:
Is this right spot to insert new 'cost model' logic - to replace the static chunking with cost_model.

Reply:
This could be a good spot, then any transactions that don't meet the model are pushed into unprocessed_txs to retry in the next slot. I don't think it really needs to mess with the batching though. It can pass however many transactions meet the model and then the MAX_NUM_TRANSACTIONS_PER_BATCH can split them into multiple batches if necessary.

Reply:
Actually I think in `transactions_from_packets` itself might be a good place. In there we are constructing the `Transaction` `Vec` so it's easier to just skip adding a Transaction to the list. In here we only have the reference to the Transaction list and we cannot change it, just copy it or use a partial view of it.

Reply:
Brilliant! I do feel it is bit too tight in regarding we can do with the transaction list here. Will look into move this to `transactions_to_packets`. Thanks!